sql/src/main/java/io/crate/metadata/doc/DocSchemaInfo.java

```diff
@@ -179,6 +176,33 @@ public TableInfo getTableInfo(String name) {
         return tables;
     }
 
+    private TableInfo getViewInfo(String name) {
+        try {
+            return viewInfoFactory.create(new TableIdent(schemaName, name), clusterService.state());
+        } catch (ResourceUnknownException e) {
+            return null;
+        }
+    }
+
+    private Collection<String> viewNames() {
+        Set<String> views = new HashSet<>();
```

Comment:
can be moved below the if statement to avoid the creation if there is no viewMetaData